bin/DESeq2.R

```diff
@@ -173,19 +190,79 @@ if (!is.null(opt$relevel)) {
 }
 
 # Run DESeq function
-cds <- DESeqDataSetFromMatrix( countData =count.table, colData =metadata, design = eval(parse(text=as.character(design[[1]]))))
-cds <- DESeq(cds,  parallel = FALSE)
+if (opt$input_type == "rawcounts") {
+    cds <- DESeqDataSetFromMatrix( countData =count.table, colData =metadata, design = eval(parse(text=as.character(design[[1]]))))
+    cds <- DESeq(cds,  parallel = FALSE)
+} else if (opt$input_type %in% c("rsem", "salmon")) {
+    ## Create a dataframe which consists of both the gene id and the transcript name
+    gtf <- rtracklayer::import(opt$gtf)
+    gtf <- as.data.frame(gtf, header=T)
+    # TODO: For some gtf files, transcript_id does not work!!
```

Comment:
I would suggest leaving the TODO as an info in case another genome needs to be implemented in the future (might work with transcript_id, but might also not work)